server/IdentityService/Services/IdentityService.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using IdentityService.Models;
+using IdentityService.Utils;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+
+namespace IdentityService.Services
+{
+    public class IdentityService : IIdentityService
+    {
+        private readonly UserContext _context;
+        private readonly IOptions<Audience> _settings;
+
+        public IdentityService(IOptions<Audience> settings, UserContext context)
+        {
+            _settings = settings;
+            _context = context;
+        }
+
+        public string Login(string login, string password)
+        {
+            var passwordHash = PasswordUtil.CreatePasswordHash(password);
+            var user = _context.Users.FirstOrDefault(u => u.Login.Equals(login) && u.PasswordHash.Equals(passwordHash));
+            if (user == null)
+            {
+                throw new NullReferenceException();
+            } 
+            return GenerateTokenFor(user);
+        }
+
+        public User CreateUser(string login, string password)
+        {
+            if (_context.Users.FirstOrDefault(u => u.Login.Equals(login)) != null)
+                throw new ArgumentException("User already exists");
+
+            var user = new User(login, password);
+            _context.Users.Add(user);
+            _context.SaveChanges();
+            return user;
+        }
+
+        public List<User> GetAccounts()
+        {
+            return _context.Users.ToList();
+        }
+
+        private string GenerateTokenFor(User user)
+        {
+            var creationTime = DateTime.Now;
+            var signingKey = new SymmetricSecurityKey(Encoding.ASCII.GetBytes(_settings.Value.Secret));
+
+            var jwt = new JwtSecurityToken(
+                _settings.Value.Iss,
+                _settings.Value.Aud,
+                CreateClaims(user, creationTime),
+                creationTime,
+                creationTime.Add(TimeSpan.FromMinutes(30)),
```

Comment:
I guess we could create a task in backlog to create a parametrization mechanism and use it here and there

Reply:
We can do it, but at the moment I don't see any other example where we can use it (or at least on current state of project :p)